### ISSUE TITLE
Chat model might work better

### ISSUE THREAD

**Issue by allenzren** [kevinliang888/IntroPlan#1]
Thanks for the codebase :)

Actually I believe now `logit_bias` does not affect the logprob outputs at all. It might be better to use the chat model instead, and instruct the model to output only the multiple choice tokens. For example:

```python
system_prompt = "You are a helpful assistant. You will be asked a multiple choice question. Please answer with a single capital letter from A, B, C, D, E, and do not output anything else."

prompt = "What is the color of a fire truck? A) Red B) Blue C) Green D) Yellow E) None of the above. Answer:"

response = client.chat.completions.create(
    model="gpt-4-turbo-2024-04-09",
    logit_bias=None,
    logprobs=True,
    top_logprobs=20, # maximum possible
    max_tokens=1,
    seed=42,
    messages=[
        {
            "role": "system",
            "content": system_prompt,
        },
        {
            "role": "user",
            "content": prompt,
        },
    ],
)
toplogprobs = response.choices[0].logprobs.content[0].top_logprobs
token_logprobs = [(t.token, t.logprob) for t in toplogprobs]
```

And then the top 20 tokens returned tend to cover A/B/C/D/E.

```python
token_logprobs: [('A', -1.9361265e-07), ('B', -16.53125), (' A', -17.53125), ('**', -18.171875), ('\n', -19.28125), ('E', -19.578125), ('C', -19.890625), ('a', -20.203125), ('D', -20.890625), ('А', -21.203125), ('>A', -21.296875), ('*A', -21.484375), ('Answer', -21.828125), ('\u200b', -22.578125), ('The', -22.671875), ('"A', -22.859375), (':A', -23.03125), ('\tA', -23.296875), ('(A', -23.328125), (' ', -23.375)]
```

**Comment by kaiquliang** [comment]
Thanks for pointing that out! We indeed used the chat model but it is a good catch that chat model can now return a maximum of 20 top log probabilities. We updated our code to return 20 tokens to cover A/B/C/D/E as many as possible. However, there might still be some cases where these tokens do not appear in the list of the top 20 tokens returned, so there's no guarantee they will always be included.

**Comment by kevinliang888** [comment]
Thanks for pointing that out! We indeed used the chat model but it is a good catch that chat model can now return a maximum of 20 top log probabilities. We updated our code to return 20 tokens to cover A/B/C/D/E as many as possible. However, there might still be some cases where these tokens do not appear in the list of the top 20 tokens returned, so there's no guarantee they will always be included.
